src/vm/classcompat.cpp

```diff
@@ -2596,9 +2596,14 @@ VOID    MethodTableBuilder::EnumerateClassMethods()
             }
         }
 
-#ifndef FEATURE_DEFAULT_INTERFACES
         // Some interface checks.
```

Comment:
You can keep this under `#ifdef FEATURE_NATIVE_IMAGE_GENERATION` to make it clear that this check kicks in under crossgen only.